source/acquaman/AMGenericStepScanConfiguration.cpp

```diff
@@ -0,0 +1,187 @@
+#include "AMGenericStepScanConfiguration.h"
+
+#include "ui/acquaman/AMGenericStepScanConfigurationView.h"
+#include "acquaman/AMGenericStepScanController.h"
+
+#include <math.h>
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(QObject *parent)
+	: AMStepScanConfiguration(parent)
+{
+	setName("Generic Scan");
+	setUserScanName("Generic Scan");
+}
+
+AMGenericStepScanConfiguration::AMGenericStepScanConfiguration(const AMGenericStepScanConfiguration &original)
+	: AMStepScanConfiguration(original)
+{
+	setName(original.name());
+	setUserScanName(original.name());
+
+	computeTotalTime();
+
+	for (int i = 0, size = scanAxes_.count(); i < size; i++){
```

Comment:
@davidChevrier @dretrex 
Yeah. You are right. `.count()` is executed for each loop.